code/_onclick/hud/ability_screen_objects.dm

```diff
@@ -134,6 +134,14 @@
 			return S
 	return null
 
+/obj/screen/movable/ability_master/proc/synch_spells_to_mind(var/datum/mind/M)
+	if(!M)
+		return
+	LAZYINITLIST(M.learned_spells)
+	for(var/obj/screen/ability/spell/screen in spell_objects)
+		var/spell/S = screen.spell
+		M.learned_spells |= S
```

Comment:
`LAZYDISTINCTADD` would let you not bother with `LAZYINITLIST` before the loop.

Reply:
True, but then again, we don't need to check if the list is null on every iteration imo.